sdk/storage/azure-storage-blob-cryptography/src/test/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlockBlobApiTests.java

```diff
@@ -1728,6 +1728,35 @@ public void encryptionDataCaseInsensitivityDownloadToFile(String newKey, Encrypt
         compareFiles(file, outFile, 0, file.length());
     }
 
+    @ParameterizedTest
+    @MethodSource("uploadAndDownloadDifferentRegionLengthSupplier")
+    public void uploadAndDownloadDifferentRegionLength(int regionLength, int dataSize) {
+        ByteBuffer data = getRandomData(dataSize);
+        beac = mockAesKey(getEncryptedClientBuilder(fakeKey, null, ENV.getPrimaryAccount().getCredential(),
+            cc.getBlobContainerUrl(), EncryptionVersion.V2)
+            .blobName(generateBlobName())
+            .blobEncryptionOptions(new BlobEncryptionOptions().setAuthenticatedRegionDataLength(regionLength))
+            .buildEncryptedBlobAsyncClient());
```

Comment:
Would it be possible to create another version of this test where we test it from the sync interface? I think as we do the migration it would be nice to have both versions of it.

Reply:
Revisiting this when decoupling sync and async client tests